### ISSUE TITLE
"to use the 'ssh' connection type with passwords or pkcs11_provider, you must install the sshpass program"

### ISSUE THREAD

**Issue by braveadmin** [ansible/awx#13510]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Bug Summary

Suddenly I started receiving this error message when AWX tries to connect to a VM:
"to use the 'ssh' connection type with passwords or pkcs11_provider, you must install the sshpass program"

I didn't change anything in the cluster running AWX, neither changed its version, so I can't understand or find the root cause of this issue.

This is the ansible.cfg being used:
```
[defaults]
host_key_checking = False
PreferredAuthentications = password
remote_user = appusr
ansible_python_interpreter=/usr/bin/python3
callbacks_enabled = ansible.posix.profile_tasks
#vault_password_file =

[privilege_escalation]
become_user = root

```

### AWX version

21.9.0

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs
- [ ] Collection
- [X] CLI
- [ ] Other

### Installation method

minikube

### Modifications

no

### Ansible version

2.14.2

### Web browser

Edge

### Steps to reproduce

Not really, it started happening suddenly without any apparent changes.

### Expected results

Connect to VM as usually and start performing tasks.

### Actual results

Task finishes with the following message:
fatal: [VM]: FAILED! => {
    "msg": "to use the 'ssh' connection type with passwords or pkcs11_provider, you must install the sshpass program"
}

**Comment by braveadmin** [comment]
Related issue: #https://github.com/ansible/awx-ee/issues/161

Workaround: Change the execution environment to `quay.io/ansible/awx-ee:21.11.0`

**Comment by mdsketch** [comment]
Can confirm this works, I started having the same issue yesterday.

Any idea when this will be fixed and I can switch back to latest? or maybe I should be specifying the versions.

**Comment by ompals92** [comment]
Change execution environment image pull path, It's working.

quay.io/ansible/awx-ee:21.11.0

**Comment by Frewx** [comment]
Can you tell me how to change this path ?

**Comment by ghost** [comment]
Go to the execution environment tab in the side bar in AWX.

Edit whatever your environment is, for me it was the default `AWX EE (latest)` 

Then just change the tag on the quay url

**Comment by relrod** [comment]
Closing as duplicate of ansible/awx-ee#161. Working on getting this and a few other things fixed up.

**Comment by koshrf** [comment]
Still missing sshpass, no idea why this was closed and not tested. And it isnt a duplicate of https://github.com/ansible/awx-ee/issues/161 that one was about the ovirt module missing and someone mentioned the sshpass part.

**Comment by relrod** [comment]
> no idea why this was closed and not tested

No need to be snarky.

> that one was about the ovirt module

You might consider re-reading the issue title in question:

> ovirt.ovirt Module not found anymore after #160 and **missing sshpass programm for Ansible SSH with Password**

That said, yes, it looks like this is still missing. I will get a patch up.

**Comment by koshrf** [comment]
> > no idea why this was closed and not tested
> 
> No need to be snarky.
> 
> > that one was about the ovirt module
> 
> You might consider re-reading the issue title in question:
> 
> > ovirt.ovirt Module not found anymore after #160 and **missing sshpass programm for Ansible SSH with Password**
> 
> That said, yes, it looks like this is still missing. I will get a patch up.

Thank you! wasn't trying to be snarky, wrong phrasing on my part, I apologize, and you are right the title of the ovirt issue has the ssh issue too but it seems the ovirt issue was fixed but not the sshpart. 

I opened another issue with similar title  https://github.com/ansible/awx-ee/issues/170 because I wasn't sure if 'closed' issues were read again.
